# Урок 1. Введение. Среда разработки и Jupyter Notebook

Сегодня мы создадим первый ноутбук, научимся запускать код и оформим расчёт как понятный мини-отчёт.

## 1. Зачем нужен Python

Ноутбук объединяет вопрос, вычисления, результат и вывод. Python важен для курса не только простым синтаксисом: его экосистема связывает данные, вычисления, визуализацию, модели и API.

Тяжёлые вычисления часто выполняют оптимизированные библиотеки или GPU, а Python управляет экспериментом. Мы изучаем его, чтобы быстро проходить путь от вопроса до проверяемого результата.

**Вопрос группе:** если AI может написать выражение, что человек всё равно должен проверить?

**Первый вопрос:** сколько стоят три товара по 1250 рублей?

In [1]:
1250 * 3

3750

**Вывод:** три товара по 1250 рублей стоят 3750 рублей.

## 2. Карта курса

Мы пройдём путь от первых выражений на Python до исследования данных и моделей машинного обучения. Каждый новый инструмент будет появляться тогда, когда нужен для вопроса.

> Код нужно запускать, результат — проверять, вывод — формулировать словами.

## 3. Создаём первый Notebook

1. Откройте JupyterLab.
2. Создайте Notebook с ядром Python.
3. Назовите файл `lesson_01_first_notebook.ipynb`.
4. Сохраните файл.

Notebook отправляет код в Kernel — отдельный процесс Python, который выполняет ячейку и возвращает результат. Notebook и Kernel — не одно и то же. Подробное состояние Kernel увидим на занятии №2 при изучении переменных.

Все задания выполняйте в собственном ноутбуке.

## 4. Code-ячейки

Code-ячейка содержит код. `Shift+Enter` запускает её и показывает результат. Перед запуском спрогнозируйте ответ, затем измените одно число и выполните ячейку повторно. Изменённый код не меняет старый результат сам.

In [2]:
1200 + 350

1550

In [3]:
800 * 4

3200

### Задание 1. Прогноз результата

1. До запуска предположите результат выражения `5000 * 0.9`.
2. Проверьте предположение в Code-ячейке.
3. Замените `5000` другой суммой и запустите ячейку снова.

In [4]:
5000 * 0.9

4500.0

## 5. Markdown-ячейки

Markdown-ячейка нужна для заголовков, условий, пояснений и выводов.

### Задание 2. Оформление

Создайте заголовок `# Мой первый Notebook`, подзаголовок `## Расчёты`, предложение о цели работы и список из двух освоенных действий.

<details>
<summary>Показать пример решения</summary>

# Мой первый Notebook

## Расчёты

Цель работы — научиться выполнять простые вычисления в Jupyter Notebook.

- Создавать Markdown- и Code-ячейки.
- Запускать ячейки и проверять результат.

</details>

## 6. Практика «Первый заказ»

### Обязательное задание

Товар стоит 1250 рублей. Покупатель заказывает 3 товара и получает скидку 10%.

Коэффициент `0.9` означает, что после скидки остаётся 90% исходной суммы. Сначала спросите: «Оба выражения работают, но на какой вопрос отвечает каждое?»

```python
1250 * 3 * 0.10
1250 * 3 * 0.90
```

В собственном ноутбуке запишите вопрос, рассчитайте стоимость до и после скидки, запишите ответ словами, затем измените цену и обновите расчёт и вывод.

In [5]:
1250 * 3

3750

In [6]:
1250 * 3 * 0.9

3375.0

<details>
<summary>Показать пример вывода</summary>

**Пример вывода:** после скидки 10% покупатель заплатит 3375 рублей.

</details>

### Если закончили раньше

Доставка стоит 300 рублей. Рассчитайте сумму со скидкой и доставкой. Затем сделайте доставку бесплатной, определите экономию и оформите оба ответа словами.

In [7]:
1250 * 3 * 0.9 + 300

3675.0

## 7. Ошибка — это информация

Посмотрим, что произойдёт при делении на ноль. Спросите: «Что полезнее: увидеть слово Error или понять, где Python объясняет причину?» Найдите в сообщении название ошибки, затем исправьте выражение.

In [8]:
# Для демонстрации ошибки временно уберите знак # в следующей строке.
# 10 / 0

In [9]:
10 / 2

5.0

## 8. Самостоятельная мини-работа

### Обязательное задание

Цена товара — 950 рублей, количество — 4, скидка — 15%. Создайте три ячейки: условие или вопрос, расчёт и вывод. Готового выражения в условии нет. Перед раскрытием решения спросите себя: «Что я теперь смогу проверить самостоятельно, даже если выражение написал AI?»

In [10]:
950 * 4 * 0.85

3230.0

### Если закончили раньше

Рассчитайте заказ со скидкой 20%, найдите разницу между двумя итоговыми суммами, объясните выгоду словами, затем измените количество товаров и обновите результаты.

In [11]:
950 * 4 * 0.85
950 * 4 * 0.80
950 * 4 * 0.85 - 950 * 4 * 0.80

190.0

## 9. Чистый запуск и домашнее задание

После урока создайте один приватный репозиторий `python-ai-course` на весь курс. Через него вы будете получать материалы, сдавать работы и получать обратную связь.

Все действия выполняются дома по отдельной инструкции. Сейчас получение данных и проверка выполняются вручную: автоматической проверки ещё нет. Пароли, токены и коды подтверждения никому передавать нельзя.

Перед завершением занятия преподаватель кратко показывает профессиональную привычку `Restart Kernel → Run All`: сохранённый вывод может быть старым, поэтому проверяют весь путь получения результата.

## Домашнее задание

1. Выберите простой товар или объект, цену, количество и скидку либо другой один простой параметр. Создайте Notebook «Мой первый расчёт»: заголовок и вопрос.
2. Выполните не менее двух расчётов, измените хотя бы одно исходное значение и повторите расчёт.
3. Сравните результаты, добавьте вывод словами и выполните все ячейки с начала.
4. По отдельной инструкции создайте приватный репозиторий `python-ai-course`.
5. Добавьте `MaiorovAlex` и `AlexBot-hub` как collaborators.
6. Загрузите ноутбук в `homework/lesson-01/solution.ipynb`.
7. Сообщите преподавателю GitHub username и ссылку способом, указанным преподавателем.

## Итог урока

Теперь вы умеете создавать и сохранять Notebook, различать Markdown- и Code-ячейки, запускать код, проверять результат, читать название простой ошибки и оформлять расчёт как небольшой отчёт. Вы также умеете спросить, на какой вопрос отвечает код, даже если его предложил AI.